src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/GlobalEventManager.cs

```diff
@@ -426,33 +427,28 @@ private static RoutedEventHandlerInfoList GetUpdatedDTypedClassListeners(
 
         #region Global Index for RoutedEvent and EventPrivateKey
 
-        internal static int GetNextAvailableGlobalIndex(object value)
+        /// <summary>
+        /// Increments the global counter for <see cref="RoutedEvent"/> and <see cref="EventPrivateKey"/> storage.
+        /// </summary>
+        /// <returns>Globally unique index for the event within the application.</returns>
+        /// <exception cref="InvalidOperationException">Thrown in case the index is bigger than <see cref="int.MaxValue"/>.</exception>
+        internal static int GetNextAvailableGlobalIndex()
         {
-            int index;
-            lock (Synchronized)
-            {
-                // Prevent GlobalIndex from overflow. RoutedEvents are meant to be static members and are to be registered 
-                // only via static constructors. However there is no cheap way of ensuring this, without having to do a stack walk. Hence 
-                // concievably people could register RoutedEvents via instance methods and therefore cause the GlobalIndex to 
-                // overflow. This check will explicitly catch this error, instead of silently malfuntioning.
-                if (_globalIndexToEventMap.Count >= Int32.MaxValue)
-                {
-                    throw new InvalidOperationException(SR.TooManyRoutedEvents);
-                }
-
-                index = _globalIndexToEventMap.Add(value);
-            }
-            return index;
-        }
-
-        // Must be called from within a lock of GlobalEventManager.Synchronized
-        internal static object EventFromGlobalIndex(int globalIndex)
-        {
-            return _globalIndexToEventMap[globalIndex];
+            // Prevent GlobalIndex from overflow. RoutedEvents are meant to be static members and are to be registered 
+            // only via static constructors. However there is no cheap way of ensuring this, without having to do a stack walk. Hence 
+            // concievably people could register RoutedEvents via instance methods and therefore cause the GlobalIndex to 
+            // overflow. This check will explicitly catch this error, instead of silently malfuntioning.
+            uint newIndex = Interlocked.Increment(ref s_globalEventIndex);
+            if (newIndex >= int.MaxValue)
+                throw new InvalidOperationException(SR.TooManyRoutedEvents);
+
+            return (int)newIndex;
         }
 
-        // must be used within a lock of GlobalEventManager.Synchronized
-        private static ArrayList _globalIndexToEventMap = new ArrayList(100); // figure out what this number is in a typical scenario
+        /// <summary>
+        /// Access must be done atomically, currently only accessed via <see cref="GetNextAvailableGlobalIndex"/> method.
```

Comment:
Could you please explain why we still need the global index even after removing the globalIndexToEventMap?